### DISCUSSION TITLE
iterate dictionary show keys format as array

### DISCUSSION THREAD

**Discussion by djasonic** [Rooyca/obsidian-api-request#36] [Q&A] [answered]
Seeking to return dictionary keys as list of strings and it was not obvious the syntax to do so nor did I find an existing issue or topic here that appeared to cover this use-case. 

```req
url: https://lucide.dev/api/icon-nodes
method: get
headers: {"Accept": "application/json"}
req-id: req-icon-nodes
save: cache/icon-nodes.json
show: {..} -> content
format: <h1>{}</h1>
disabled
```

Where the returned body is a dictionary of objects and the key is the name I need but do not know up front and I am not interested in the dictionary values.  

```json

{
    "a-large-small": [
        [
            "path",
            {
                "d": "M21 14h-5"
            }
        ]
    ],
    "a-arrow-up": [
        [
            "path",
            {
                "d": "M3.5 13h6"
            }
        ]
    ],

...
...

    "sunset": [
        [
            "path",
            {
                "d": "M12 10V2"
            }
        ]
    ]
}
```

To return a list of icon names from those keys

a-large-small
a-arrow-up
...
...
sunset

**Comment by Rooyca** [comment]
Hello @djasonic,

What you are looking for cannot be done... for now. 

~~~markdown
```req
url: https://lucide.dev/api/icon-nodes
req-id: req-icon-nodes
save: cache/icon-nodes.json
show: json
format: <h1>{}</h1>
render
disabled
```
~~~

Here we are using `show` to get the json from the response, I believe this need to be done because the saved response stores all (headers, status and json), but we cannot get the keys because for that we would need something like `Object.keys(jsonObject)` and we don't have that option. Maybe you could use [dataview](https://blacksmithgu.github.io/obsidian-dataview/) to read the `localStorage` and get the keys using `JS`.

**Reply by Rooyca** [reply, marked as answer]
I'm adding a new option to `show`, it's called `{gk}` (stands for `get keys`) to make this process a lot easier. It should be UP in a bit. Here is a working example:

~~~markdown
```req
url: https://lucide.dev/api/icon-nodes
req-id: req-icon-nodes
save: cache/icon-nodes.json
show: json -> {gk}
disabled
format: <h1>{}</h1>
render
```
~~~

**Reply by Rooyca** [reply]
@djasonic it should be working on the new update :)

have a good day.

**Reply by djasonic** [reply]
Yes I am seeing the list render now.  Thank you this is very useful.

However an issue noticed today where the raw json appears about midway through the list of keys.
